docs/markdown/Helm/helm-deployments.md

```diff
@@ -0,0 +1,197 @@
+> It's good practice to publish your Docker images using tags other than `latest` and Pants preferred behaviour is to choose those are this guarantees that the _version_ of the Docker image being deployed is the expected one.
+
+Value override files
+--------------------
+
+As seen in the initial examples of the usage of the `helm_deployment` target, it can be seen that some of the source value files used the word `override` in the file name.
+
+When using deployments that may have more than one YAML file as the source of configuration values, it is recommended use the `override` word in those filenames that are meant to override values coming from other more common or general sources.
+
+Besides, you may be interested in organizing your deployment source files using a nested folder structure as the following:
+
+```
+src/deployment/config_maps.yaml
+src/deployment/services.yaml
+src/deployment/dev/services.yaml
+src/deployment/uat/services.yaml
+```
+
+```text src/deployment/BUILD
+helm_deployment(name="dev", dependencies=["//src/chart"], sources=["*.yaml", "dev/*.yaml"])
+
+helm_deployment(name="uat", dependencies=["//src/chart"], sources=["*.yaml", "uat/*.yaml"])
+```
+
+In this case, files that are in nested folders will also act as value override files.
+
+We believe that this approach gives enough flexibility to organise your deployment sources as you better prefer while providing an intuitive way to Pants' users to ensure that the list of value file sources is in the right order every single time.
+
+In addition to value files, you can also use inline values in your `helm_deployment` targets by means of the `values` field. All inlines values that are set this way will override any entry that may come from value files, whether the source file was an override file or not. The relationship between the three scopes can be imagined as follows:
+
+```
+Value files < Value override files < Inline values
+```
+
+Deploying
+---------
+
+Continuing with the example in the previous section, we can deploy it into Kubernetes using the command `./pants experimental-deploy src/deployment`. This will go trigger the following steps:
+
+1. Analyse the dependencies of the given deployment.
+2. Build and publish any first-party Docker image and Helm charts that are part of those dependencies.
+3. Post-process the Kubernetes manifests generated by Helm by replacing all references to first-party Docker images by their real final registry destination.
+4. Initiate the deployment of the final Kubernetes resources resulting from the post-processing.
+
+The `experimental-deploy` goal also supports default Helm pass-through arguments that allow to change the deployment behaviour to be either atomic or a dry-run or even what is the Kubernetes config file and target context to be used in the deployment.
```

Comment:
By "kubernetes config file" do you mean the helm values file?

Reply:
I meant the `kubeconfig` file (should be more specific about it) which can be specified to Helm using the `--kubeconfig` argument.

In CI it's quite probable that the file may live in a non-standard place (it does in our case) so it is useful to be able to pass that to the underlying Helm process.